src/service.rs

```diff
@@ -981,23 +911,21 @@ where
             match target {
                 TargetProtocol::All => {
                     self.protocol_configs
-                        .keys()
-                        .for_each(|name| session.open_proto_stream(name));
+                        .values()
+                        .for_each(|meta| session.open_proto_stream(&meta.name()));
                 }
                 TargetProtocol::Single(proto_id) => {
-                    self.protocol_configs
-                        .values()
-                        .find(|meta| meta.id() == proto_id)
-                        .and_then(|meta| {
-                            session.open_proto_stream(&meta.name());
-                            Some(())
-                        });
+                    self.protocol_configs.get(&proto_id).and_then(|meta| {
```

Comment:
please use `if let`, no need to return an empty `Some(())`:

```suggestion
                    if let Some(meta) = self.protocol_configs.get(&proto_id) {
                        session.open_proto_stream(&meta.name());
                    }
```